src/test/java/io/kestra/plugin/confluence/pages/UpdateConfluencePageTest.java

```diff
@@ -0,0 +1,140 @@
+package io.kestra.plugin.confluence.pages;
+
+import io.kestra.core.junit.annotations.KestraTest;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.runners.RunContextFactory;
+import jakarta.inject.Inject;
+import kong.unirest.HttpRequestWithBody;
+import kong.unirest.HttpResponse;
+import kong.unirest.JsonNode;
+import kong.unirest.RequestBodyEntity;
+import kong.unirest.Unirest;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.argThat;
+
+@KestraTest
+class UpdateConfluencePageTest {
+    @Inject
+    private RunContextFactory runContextFactory;
+
+    @Test
+    void missingRequiredProperties_throws() {
+        RunContext runContext = runContextFactory.of(Map.of());
+
+        // serverUrl is required -> should throw
+        UpdateConfluencePage task = UpdateConfluencePage.builder()
+            .username(Property.ofValue("user@example.com"))
+            .apiToken(Property.ofValue("token"))
+            .pageId(Property.ofValue("123"))
+            .status(Property.ofValue("current"))
+            .title(Property.ofValue("Title"))
+            .markdown(Property.ofValue("# Hello"))
+            .versionInfo(Property.ofValue(UpdateConfluencePage.VersionInfo.builder()
+                .number(Property.ofValue(2))
+                .message(Property.ofValue("msg"))
+                .build()))
+            .build();
+
+        assertThrows(IllegalArgumentException.class, () -> task.run(runContext));
+    }
+
+    @Test
+    void non2xxResponse_throwsIllegalStateException() {
+        RunContext runContext = runContextFactory.of(Map.of());
+
+        UpdateConfluencePage task = UpdateConfluencePage.builder()
+            .serverUrl(Property.ofValue("https://example.com"))
+            .username(Property.ofValue("user@example.com"))
+            .apiToken(Property.ofValue("token"))
+            .pageId(Property.ofValue("123"))
+            .status(Property.ofValue("current"))
+            .title(Property.ofValue("Title"))
+            .markdown(Property.ofValue("# Hello"))
+            .versionInfo(Property.ofValue(UpdateConfluencePage.VersionInfo.builder()
+                .number(Property.ofValue(2))
+                .message(Property.ofValue("msg"))
+                .build()))
+            .build();
+
+        HttpRequestWithBody requestWithBodyMock = Mockito.mock(HttpRequestWithBody.class);
+        RequestBodyEntity requestBodyEntityMock = Mockito.mock(RequestBodyEntity.class); 
+        HttpResponse<JsonNode> responseMock = Mockito.mock(HttpResponse.class);
+
+        try (MockedStatic<Unirest> unirest = Mockito.mockStatic(Unirest.class)) {
```

Comment:
OK got it. Just remove the UniRest dependency please.
Instead, could you create a new `docker-compose-ci.yml` (not update the existing `docker-compose.yml`) and use the [Confluence docker image](https://hub.docker.com/r/atlassian/confluence) please?
Then you can use this service by configuring it in `.github/setup-unit.sh` (used by the CI for the tests) but also by you locally to check.
If need you can also set a `.github/cleanup-unit.sh` to cleanup the env after the tests. 
Thanks!

Reply:
What about this one @kyounghunJang? :pray: